pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -258,10 +254,10 @@ func (os *OpenStack) DetachDisk(instanceID, volumeID string) error {
 	if err != nil {
 		return err
 	}
-	if volume.Status != VolumeInUseStatus {
-		errmsg := fmt.Sprintf("can not detach volume %s, its status is %s.", volume.Name, volume.Status)
-		glog.Errorf(errmsg)
-		return errors.New(errmsg)
+	if volume.Status == VolumeAvailableStatus {
```

Comment:
Yeah, I should restore the original codes, thank for your comment, I will update it soon.